auth/src/handlers/create_account.rs

```diff
@@ -4,7 +4,98 @@ pub struct CreateAccount {
     email: String,
     name: String,
     hashed_password: String,
-    accept_terms: bool,
+}
+
+impl CreateAccount {
+    fn to_provider_data(&self) -> Vec<ft_sdk::auth::UserData> {
+        vec![
+            ft_sdk::auth::UserData::Email(self.email.clone()),
+            ft_sdk::auth::UserData::Name(self.name.clone()),
+            ft_sdk::auth::UserData::Identity(self.name.clone()),
+            ft_sdk::auth::UserData::Custom {
+                key: "hashed_password".to_string(),
+                value: self.hashed_password.clone().into(),
+            },
+        ]
+    }
+
+    /// create relative path to the confirm account route handler
+    ///
+    /// this route needs to be prefixed with the mount point that you used in
+    /// your fastn app's url-mappings
+    fn create_conf_path(
+        &self,
+        conn: &mut ft_sdk::Connection,
+        user_id: ft_sdk::auth::UserId,
+    ) -> Result<String, ft_sdk::auth_provider::AuthError> {
+        let key = CreateAccount::generate_key(64);
+
+        let data = vec![ft_sdk::auth::UserData::Custom {
+            key: "conf_code".to_string(),
+            value: key.clone().into(),
+        }];
+
+        // save the conf link for later use
+        ft_sdk::auth_provider::authenticate(
+            conn,
+            auth::PROVIDER_ID,
+            &self.name,
+            data,
+            Some(user_id),
+        )?;
+
+        Ok(CreateAccount::confirmation_link(key))
+    }
+
+    fn confirm_account_html(name: &str, link: &str) -> String {
+        // TODO: until we figure out email templates, this has to do
+        format!(
+            r#"
+            <html>
+                <head>
+                    <title>Confirm your account</title>
+                </head>
+                <body>
+                    <h1>Hi {name},</h1>
+                    <p>Click the link below to confirm your account</p>
+                    <a href="{link}">Confirm your account</a>
+
+                    In case you can't click the link, copy and paste the following link in your browser:
+                    <br>
+                    <a href="{link}">{link}</a>
+                </body>
+            </html>
+            "#,
+            name = name,
+            link = link,
+        )
+    }
+
+    fn generate_key(length: usize) -> String {
+        let mut rng = rand::thread_rng();
+        rand::distributions::DistString::sample_string(
+            &rand::distributions::Alphanumeric,
+            &mut rng,
+            length,
+        )
```

Comment:
How is this working in wasm context? I think we have to expose (or we have already exposed) some host function to get random number.